File: matriz.js
//declarar matriz 3x3 del 1 al 9

console.log("---- matriz 3x3 del 1 al 9 ----")

let vector1 = [1,2,3];
let vector2 = [4,5,6];
let vector3 = [7,8,9];

let matriz = [vector1,vector2,vector3];

for (let i = 0; i < matriz.length; i++){
  console.log(matriz[i]);
  
}

//recorrido por columnas

console.log("---- Recorrido por columnas ----");

for (let j = 0; j < matriz[0].length; j++) {
  let columna = [];
  for (let i = 0; i < matriz.length; i++) {
    columna.push(matriz[i][j]);
  }
  console.log("Elementos de la columna " + (j + 1) + ":", columna.join(', '));
}

//sumar todos los elementos

  console.log("---- suma de todos los elementos ----")

  let sumaTotal = 0

  for (let i = 0; i < matriz.length; i++) {
    for (let j = 0; j < matriz[i].length; j++) {
      sumaTotal += matriz[i][j];
    }
    
  }

  console.log("la suma total de los elementos es:  ", sumaTotal);

  // intercambiar filas

  console.log("---- intercambiar la promera fila con la última ----")

  const primeraFila = matriz[0];

  matriz[0] = matriz[matriz.length - 1];

  matriz[matriz.length - 1] = primeraFila;

  console.log("---- matriz con filas intercambiadas:");
  for (let i = 0; i < matriz.length; i++){
  console.log(matriz[i]);
  
}

File: matriz.py
#matriz 3x3 con numeros del 1 al 9

matriz = [
    [1, 2, 3],
    [4, 5, 6],
    [7, 8, 9]
]

for row in matriz:
    print(row)

#recorrer por columnas 

print("Recorrido por columnas:")

for j in range(len(matriz[0])):
    for i in range(len(matriz)):
        print(matriz[i][j], end=" ")
    print()

#suma de todos los elementos

suma_total = 0
for fila in matriz:
    for elemento in fila:
        suma_total += elemento

print(f"La suma de todos los elementos es: {suma_total}")

# Intercambio de la primera fila con la última

print(f"Matriz original: {matriz}")
for row in matriz:
    print(row)

matriz[0], matriz[-1] = matriz[-1], matriz[0]


print(f"Matriz después del intercambio: {matriz}")

for row in matriz:
    print(row)
